### DISCUSSION TITLE
Transformer ignored in outgoing message

### DISCUSSION THREAD

**Discussion by bartdewulf** [nextgenhealthcare/connect#4490] [Q&A] [answered]
Hello, 

Relatively new to NextGen Connect, but have some strange behaviour I cannot understand. 
I have a message builder transformer set up on my Channel source. I need to replace the appointment statusses I receive with our own appointment states. 

So I have the following message builder step set up: 

<img width="819" alt="Screenshot 2021-04-13 at 16 20 37" src="https://user-images.githubusercontent.com/76184865/114568179-34bef000-9c74-11eb-9134-3c4a945d1ff6.png">

When I look at my encoded message (HL7v2 on the source), then the replacement has worked fine. 

At my destination (converting from v2 to v3), I see that in the RAW message the replacement is still there. So SCH.25.1 has the correct value. 
But on my encoded message of the destination, I see the original value again (from before the transformation). And I don't understand how the destination can know about this original value, since it has been modified at the source level.... 

Probably something small I forgot to set? 
Thx for the help!

**Comment by bartdewulf** [comment, marked as answer]
My mistake: I had an outbound message template in my destination channel. Even without any active transformer steps, it was always sending the message I had pasted in as a tempate. 
Removed the outbound message template, and all is working fine now. 
Will keep this discussion topic maybe for others in the future.

**Comment by ruizhannia** [comment]
I'm having a similar issue using Javascript to transform an XML to HL7.v2. I am not getting any errors on the transformer, but it is not pulling the values from the XML. If I hardcode a value then the value does populate into the HL7. Any suggestions would be helpful. I've tried using the templates and still not able to get the transformer to work.

**Reply by brentmoen** [reply]
@ruizhannia Please open a separate discussion Q&A topic with more details/screenshots/etc, this will help community members to help troubleshoot
